pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1592,21 +1609,71 @@ func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*ne
 			}
 			// increment potential IP
 			potentialDefaultIP = incrementIPForDefaultCIDR(potentialDefaultIP)
-
-			// check for overlap
-			// if it overlaps continue with the loop
+			//check for overlap
+			// if it overlaps continue
+			//no need for boolean check as if cluster vpc fails it wont get this far
 			if clusterNet.Contains(defaultNet.IP) || defaultNet.Contains(clusterNet.IP) {
 				continue
 			}
+
+			if foundPodNet {
+				if podCIDR.Contains(defaultNet.IP) || defaultNet.Contains(podCIDR.IP) {
+					continue
+				}
+
+			}
+
+			if foundServiceNet {
+				if serviceCIDR.Contains(defaultNet.IP) || defaultNet.Contains(serviceCIDR.IP) {
+					continue
+				}
+
+			}
+
 			// return the first available option that does not overlap
 			return defaultNet, nil
 		}
 	}
+
 	// if the loop finishes, it means that we have gone through all available cidr blocks in the available ranges in aws
 	// return an error that cro was unable to find an option
 	return nil, errorUtil.New("could not find a default cidr block")
 }
 
+func getPodCIDR(networkConf *v12.Network) (*net.IPNet, bool, error) {
+	var podCIDR *net.IPNet
```

Comment:
Nitpick - the var name, `podCIDR` is actually an `IPNet` object. So maybe something like `podNet` would make more sense here.